### PR TITLE
fix(bannedNodes): restrict the rule to named functions

### LINKED ISSUE #13: Internal error: Expected instance of PhpParser\Node\Name for $node->name, PhpParser\Node\Expr\ArrayDimFetch given
Hi, I got this weird error when trying to use your package.

The result of the debug is this:

```
Note: Using configuration file <path>/phpstan.neon.
<path>/src/Mapado/TicketingBundle/Export/ProTicketListGenerator.php
PHP Fatal error:  Uncaught RuntimeException: Expected instance of PhpParser\Node\Name for $node->name, PhpParser\Node\Expr\ArrayDimFetch given in <path>/vendor/ekino/phpstan-banned-code/src/Rules/BannedNodesRule.php:64
Stack trace:
#0 <path>/vendor/phpstan/phpstan/src/Analyser/Analyser.php(154): Ekino\PHPStanBannedCode\Rules\BannedNodesRule->processNode(Object(PhpParser\Node\Expr\FuncCall), Object(PHPStan\Analyser\Scope))
#1 <path>/vendor/phpstan/phpstan/src/Analyser/NodeScopeResolver.php(1691): PHPStan\Analyser\Analyser->PHPStan\Analyser\{closure}(Object(PhpParser\Node\Expr\FuncCall), Object(PHPStan\Analyser\Scope))
#2 <path>/vendor/phpstan/phpstan/src/Analyser/NodeScopeResolver.php(1139): PHPStan\Analyser\NodeScopeResolver->callNodeCallbackWithExpression(Object(Closure), Object(PhpParser\Node\Expr\FuncCall), Object(PHPStan\Analyser\Scope), Object(PHPStan\Analyser\ExpressionContext))
#3 <path>/vendor/phpstan/phpstan/s in <path>/vendor/ekino/phpstan-banned-code/src/Rules/BannedNodesRule.php on line 64
```

I do not know where it comes from, I try to parse a 1500 lines file :/

### PR DESCRIPTION
**Pull request by nanofelis**
The following cases of method calls were failing :
* $array\[$i\]()
* (new Item())()

Fixes #13

### CODE REVIEW

**Comment by szepeviktor** [comment]
@nanofelis If you include `Fixes #13` in the top comment, GitHub automatically closes that issue when this PR gets merged.

File: tests/Rules/BannedNodesRuleTest.php [hunk 1]
```diff
@@ -95,6 +96,10 @@ public function testProcessNodeWithFunctions(): void
         $node = new FuncCall(new Variable('myClosure'));
 
         $this->assertCount(0, $this->rule->processNode($node, $this->scope));
+
+        $node = new FuncCall(new Expr\ArrayDimFetch(new Variable('myArray'), LNumber::fromString('0', ['kind' => LNumber::KIND_DEC])));
```
Comment (spawnia) [inline tests/Rules/BannedNodesRuleTest.php]:
```suggestion
        $node = new FuncCall(
            new Expr\ArrayDimFetch(
                new Variable('myArray'),
                LNumber::fromString('0', ['kind' => LNumber::KIND_DEC]
            )
        )
    );
```

**Comment by merlindiavova** [comment]
Hey,

Any update on this?

**Comment by mremi** [comment]
Hi @nanofelis , sorry for the delay and thanks for this contribution.
Could you please update the CHANGELOG.md?

**Comment by spawnia** [comment]
> Hi @nanofelis , sorry for the delay and thanks for this contribution.
> Could you please update the CHANGELOG.md?

I am currently blocked by this issue and would appreciate it being merged. I suggest the following line for the `CHANGELOG.md`:

```md
- Apply the BannedNodesRule only to named functions
```

**Comment by nanofelis** [comment]
Done @mremi, @spawnia.

You're welcome !

(sorry for the delay as well, it slipped my mind)

**Comment by mremi** [comment]
Thanks 👍

**Pull request merged**